### PR TITLE
RPRBLND-1813: World support

### PR DESCRIPTION
**Pull request by bnagirniak**
### PURPOSE
World export implementation is outdated and was working in draft mode. We need more complex support of world nodes, make it work in USD nodes and in viewport.

### EFFECT OF CHANGE
Improved worlds export system with better world nodes support, make it work in USD nodes and available to change in viewport.

### TECHNICAL STEPS
1. Created module export/world:
    - added node_parser.py with NodeItem and NodeParser classes, which parses nodes in world node_tree
    - reimplemented WorldData class, adjusted sync(), implemented sync_update() in export/world/\_\_init__.py
    - implemented parsing of useful nodes used in world node_tree.
    - removed export/world.py
2. Added world syncing to BlenderDataNode USD node, made code improvements.
3. Added world syncing to ViewportEngineScene, made code improvements
4. Added utils/image.py with cache_image_file()

### NOTES FOR REVIEWERS
Viewport shading world doesn't implemented here, plan to do it in separate task.

### CODE REVIEW

**Review by VascoPi — APPROVED** [review]
Tested, color, intensity, texture works well.

File: src/hdusd/export/world/nodes.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        super().__init__(world, node, None, **kwargs)
+
+    def export(self):
+        return self.get_input_link('Surface')
+
+
+class ShaderNodeBackground(NodeParser):
+    def export(self):
+        color = self.get_input_value('Color').data
+        strength = self.get_input_scalar('Strength').data
```
Comment (DagerD) [inline src/hdusd/export/world/nodes.py:29]: redundant variables

File: src/hdusd/export/world/__init__.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        # d is dict here
+
+        intensity = d.get('intensity', 1.0)
+        if isinstance(intensity, tuple):
+            intensity = intensity[0]
+
+        data.intensity = intensity
+
+        color = d.get('color')
+        if color:
```
Comment (DagerD) [inline src/hdusd/export/world/__init__.py:80]: Do we need to check for certain variants in "if" branches and something unknown in both "else" statements and add log for it?

File: src/hdusd/export/world/__init__.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            return data
+
+        from .nodes import ShaderNodeOutputWorld
+
+        node_parser = ShaderNodeOutputWorld(world, output_node)
+        node_item = node_parser.export()
+        if not node_item:
+            return data
+
+        d = node_item.data
```
Comment (DagerD) [inline src/hdusd/export/world/__init__.py:61]: I would prefer something more understandable, e.g. "color_data", because it's used in several lines.

**Review by DagerD — COMMENTED** [review]
Commented in code

File: src/hdusd/export/world/__init__.py (hunk 2 above)
Comment (bnagirniak) [reply src/hdusd/export/world/__init__.py]: No, this is ok here

File: src/hdusd/export/world/nodes.py (hunk 1 above)
Comment (bnagirniak) [reply src/hdusd/export/world/nodes.py:29]: This done in such way to be compatible with code style in other nodes.

**Review by DagerD — APPROVED** [review]
Tested. Works ok.

**Pull request merged**